addons/payment/controllers/portal.py

```diff
@@ -127,6 +127,9 @@ def payment_pay(
         # Generate a new access token in case the partner id or the currency id was updated
         access_token = payment_utils.generate_access_token(partner_sudo.id, amount, currency.id)
 
+        # The invoice status is required to warn the user/customer about cancelled invoices
+        invoice_state = request.env["account.move"].sudo().browse(invoice_id).state
+
```

Comment:
* Add a check before to verify that the invoice exists if provided (seems we forgot that in the payment apocalypse)
```python
        warning = ''
        if invoice_id:
            invoice_sudo = request.env['account.move'].browse(currency_id).exists()
           if not invoice_sudo:
                raise ValidationError(_("The provided parameters are invalid."))
           if invoice_sudo.state == 'cancel':
               warning = _("Attention: this invoice has been cancelled.")
```